lib/panoptes/client/authentication.rb

```diff
@@ -0,0 +1,69 @@
+require 'deprecate'
+
+module Panoptes
+  class Client
+    module Authentication
+      extend Gem::Deprecate
+
+      def jwt_payload
+        raise NotLoggedIn unless @auth[:token]
+        payload, = JWT.decode @auth[:token], jwt_signing_public_key, algorithm: 'RS512'
+        payload
+      rescue JWT::ExpiredSignature
+        raise AuthenticationExpired
+      end
+
+      def token_contents
+        jwt_payload.fetch('data')
+      end
+
+      def token_expiry
+        jwt_payload.fetch('exp')
+      end
+
+      def authenticated?
+        not token_contents['id'].nil?
+      end
+
+      def authenticated_user_login
+        raise NotLoggedIn unless authenticated?
```

Comment:
I really don't see why one checks for the authenticated state of the payload on each attribute access. 

The code path currently is through`token_contents` which calls `jwt_payload`, in turn this decodes the JWT and verifies the integrity of the singing code (via private key). If this decoding is successfull the token is valid and will conform to the encoded payload format that is set in https://github.com/zooniverse/Panoptes/blob/d768539038779301e42ac4255e98407917bf8cc1/config/initializers/doorkeeper.rb#L71

If we can decode the token then it's trust worthy and there is no need to check that it has an 'id' data payload key.

Please correct my understanding if it's wrong.